ddl/table.go

```diff
@@ -190,11 +228,25 @@ func onCreateView(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 	}
 }
 
-func onDropTableOrView(t *meta.Meta, job *model.Job) (ver int64, _ error) {
+func onDropTableOrView(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	tblInfo, err := checkTableExistAndCancelNonExistJob(t, job, job.SchemaID)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
+	var physicalTableIDs = []int64{tblInfo.ID}
+	if tblInfo.Partition != nil {
+		for _, p := range tblInfo.Partition.Definitions {
+			physicalTableIDs = append(physicalTableIDs, p.ID)
+		}
+	}
+	syncPlacementRules := func() error {
+		err = dropRuleBundles(d, physicalTableIDs)
```

Comment:
Maybe we should not drop bundles when dropping table. We may still expect the dropped table data follows the constraints even they table is dropped. For example the compliance case or the stale read case.